File: openmrs-client/src/main/java/org/openmrs/mobile/activities/patientdashboard/diagnosis/PatientDiagnosisFragment.java
/*
 * The contents of this file are subject to the OpenMRS Public License
 * Version 1.0 (the "License"); you may not use this file except in
 * compliance with the License. You may obtain a copy of the License at
 * http://license.openmrs.org
 *
 * Software distributed under the License is distributed on an "AS IS"
 * basis, WITHOUT WARRANTY OF ANY KIND, either express or implied. See the
 * License for the specific language governing rights and limitations
 * under the License.
 *
 * Copyright (C) OpenMRS, LLC.  All Rights Reserved.
 */

package org.openmrs.mobile.activities.patientdashboard.diagnosis;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.Menu;
import android.view.MenuInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ArrayAdapter;
import android.widget.ListView;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

import org.openmrs.mobile.R;
import org.openmrs.mobile.activities.patientdashboard.PatientDashboardActivity;
import org.openmrs.mobile.activities.patientdashboard.PatientDashboardContract;
import org.openmrs.mobile.activities.patientdashboard.PatientDashboardFragment;

import java.util.List;

public class PatientDiagnosisFragment extends PatientDashboardFragment implements PatientDashboardContract.ViewPatientDiagnosis {

    private ListView mDiagnosisList;
    private PatientDashboardActivity mPatientDashboardActivity;

    public static PatientDiagnosisFragment newInstance() {
        return new PatientDiagnosisFragment();
    }

    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        super.setPresenter(mPresenter);
        setHasOptionsMenu(true);
    }

    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        View fragmentLayout = inflater.inflate(R.layout.fragment_patient_diagnosis, null, false);
        mDiagnosisList = fragmentLayout.findViewById(R.id.patientDiagnosisList);
        TextView emptyList = fragmentLayout.findViewById(R.id.emptyDiagnosisListView);
        mDiagnosisList.setEmptyView(emptyList);
        return fragmentLayout;
    }

    @Override
    public void onCreateOptionsMenu(Menu menu, MenuInflater inflater) {
        // This method is intentionally empty
    }

    @Override
    public void setDiagnosesToDisplay(List<String> encounters) {
        ArrayAdapter<String> adapter =
                new ArrayAdapter<>(getActivity(), android.R.layout.simple_list_item_1, encounters);
        mDiagnosisList.setAdapter(adapter);
    }

    @Override
    public void setUserVisibleHint(boolean isVisibleToUser) {
        super.setUserVisibleHint(isVisibleToUser);
        if (isVisibleToUser) {
            try {
                mPatientDashboardActivity.hideFABs(true);
            } catch (Exception e) {
                e.printStackTrace();
            }
        }
    }
}

File: openmrs-client/src/main/java/org/openmrs/mobile/activities/patientdashboard/diagnosis/PatientDiagnosisFragment.kt
/*
 * The contents of this file are subject to the OpenMRS Public License
 * Version 1.0 (the "License"); you may not use this file except in
 * compliance with the License. You may obtain a copy of the License at
 * http://license.openmrs.org
 *
 * Software distributed under the License is distributed on an "AS IS"
 * basis, WITHOUT WARRANTY OF ANY KIND, either express or implied. See the
 * License for the specific language governing rights and limitations
 * under the License.
 *
 * Copyright (C) OpenMRS, LLC.  All Rights Reserved.
 */
package org.openmrs.mobile.activities.patientdashboard.diagnosis

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ArrayAdapter
import androidx.core.os.bundleOf
import androidx.fragment.app.viewModels
import androidx.lifecycle.Observer
import com.openmrs.android_sdk.library.models.Result
import com.openmrs.android_sdk.utilities.ApplicationConstants.BundleKeys.PATIENT_ID_BUNDLE
import dagger.hilt.android.AndroidEntryPoint
import org.openmrs.mobile.activities.BaseFragment
import org.openmrs.mobile.activities.patientdashboard.PatientDashboardActivity
import org.openmrs.mobile.databinding.FragmentPatientDiagnosisBinding

@AndroidEntryPoint
class PatientDiagnosisFragment : BaseFragment() {
    private var _binding: FragmentPatientDiagnosisBinding? = null
    private val binding get() = _binding!!

    private val viewModel: PatientDashboardDiagnosisViewModel by viewModels()

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View {
        _binding = FragmentPatientDiagnosisBinding.inflate(inflater, null, false)

        setupListView()
        setupObserver()
        fetchDiagnoses()

        return binding.root
    }

    private fun setupListView() {
        binding.patientDiagnosisList.emptyView = binding.emptyDiagnosisListView
    }

    private fun setupObserver() {
        viewModel.result.observe(viewLifecycleOwner, Observer { result ->
            if (result is Result.Success) showDiagnosesList(result.data)
        })
    }

    private fun fetchDiagnoses() {
        viewModel.fetchDiagnoses()
    }

    private fun showDiagnosesList(diagnoses: List<String>) {
        val adapter = ArrayAdapter(requireActivity(), android.R.layout.simple_list_item_1, diagnoses)
        binding.patientDiagnosisList.adapter = adapter
    }

    override fun onDestroyView() {
        super.onDestroyView()
        _binding = null
    }

    companion object {
        fun newInstance(patientId: String): PatientDiagnosisFragment {
            val fragment = PatientDiagnosisFragment()
            fragment.arguments = bundleOf(Pair(PATIENT_ID_BUNDLE, patientId))
            return fragment
        }
    }
}
